clean up imports and add missing funcs

ANaka opened the pull request:
## Purpose

This PR is meant to stay within the cellular-longevity fork of hydra-zen. The purpose is to get the `add_conf` and `ConfMode` functionality working with the latest version of hydra zen

## To-do
- [ ] Read [previous implementation of `like`](https://github.com/mit-ll-responsible-ai/hydra-zen/pull/219) and articulate how we differ from that if we do. See also [here](https://github.com/mit-ll-responsible-ai/hydra-zen/issues/205#issuecomment-1017951452)
- [ ] Find a workaround for the deprecation of `builds_bases`. Most important item is to recapitulate the functionality of the `__call__` method in our `ZenExtras` base class
  -  Our `ZenExtras` was pretty loaded with features; may make sense to strip that down or create a simpler version
- [ ] Examine the new [zen store functionality](https://github.com/mit-ll-responsible-ai/hydra-zen/pull/331) and determine whether to replace the `.store` methods that we put into our `ZenExtras`
- [ ] Build out test suite

shababo commented:
> * Find a workaround for the deprecation of `builds_bases`. Most important item is to recapitulate the functionality of the `__call__` method in our `ZenExtras` base class

Looks like we didn't even need to use `make_custom_builds_fn` and can just use `builds` directly, e.g.:
`conf = builds(wrapped_cls, populate_full_signature=True, builds_bases=(ZenExtras,), zen_meta=zen_meta_defaults, **kwargs)`

ANaka commented on src/hydra_zen/structured_configs/_add_conf.py:
```diff
@@ -255,53 +304,12 @@ def check_signature_has_defaults_for_all_parameters(cls_or_func):
 
 @dataclass
-class BatteriesIncludedConf:
+class ZenExtras:
```
we should talk about this, it looks like some of the changes here are undoing some purposeful refactoring I put in place

rsokl commented:
Hello, I just stumbled across this fork -- there are some neat ideas here! Don't hesitate to open Discussion/Issue threads on the main hydra-zen repo if you'd ever like to chat about how hydra-zen can better facilitate your work 😄 . I'm keen to understand the ways people are using the library so that we add high-impact features. (But also no worries if you just want to stay focused on your fork)

> Find a workaround for the deprecation of builds_bases

Just FYI, hydra-zen 0.9.0 made it possible to effectively specify `builds_bases` in `make_custom_builds_fn` again, but via `zen_dataclass`

```python
from hydra_zen import make_custom_builds_fn, builds, to_yaml

Parent = builds(dict, x=1)
cbuilds = make_custom_builds_fn(zen_dataclass={'bases': (Parent,)})
```

Cheers!

ANaka commented:
>Don't hesitate to open Discussion/Issue threads on the main hydra-zen repo if you'd ever like to chat about how hydra-zen can better facilitate your work smile .

@rsokl - thanks for reaching out - we were planning on doing exactly that! We built pinned to 0.7 so our thought was to get stuff a little more up to date with the latest version first. I think we have the piece with `builds_bases`/`zen_dataclass` sorted out - thanks for the FYI. It looks to me like some of the things we were doing with `ConfigStore` might be redundant with what `ZenStore` is doing and so I'd like to look at that a little more and do any necessary reconciliation. We'll open a thread once that's handled!

ANaka commented:
I reverted back to some of the refactoring I did, mainly the bits that split `ZenExtras` into two classes. `Conf`/`@add_conf` is a simplified version of it that only retains the basic functionality (attaching a configuration dataclass to decorated classes) while `BatteriesIncludedConf`/`@add_batteries` has all the convenience methods included as well. (open to suggestion on naming)

I also updated the notebook you added - the basic interaction with `ConfMode` works if you use `@add_batteries` now. Will look at reconciling the tests later. 

>Looks like we didn't even need to use make_custom_builds_fn and can just use builds directly, e.g.:
conf = builds(wrapped_cls, populate_full_signature=True, builds_bases=(ZenExtras,), zen_meta=zen_meta_defaults, **kwargs)

@shababo good callout - I think that makes sense and kept this change.
